recipes/imagemagick/all/conanfile.py

```diff
@@ -0,0 +1,516 @@
+            command.append(msvc_version)
+            command.append("/hdri" if self.options.hdri else "/noHdri")
+            command.append("/Q%s" % self.options.quantum_depth)
+            if self.settings.arch == "x86_64":
+                command.append("/x64")
+            command = " ".join(command)
+
+            self.output.info(command)
+            self.run(command, run_environment=True)
+
+        # disable incorrectly detected OpenCL
+        baseconfig = os.path.join(
+            self._source_subfolder, "MagickCore", "magick-baseconfig.h"
+        )
+        tools.replace_in_file(
+            baseconfig,
+            "#define MAGICKCORE__OPENCL",
+            "#undef MAGICKCORE__OPENCL",
+            strict=False,
+        )
+        tools.replace_in_file(
+            baseconfig,
+            "#define MAGICKCORE_HAVE_CL_CL_H",
+            "#undef MAGICKCORE_HAVE_CL_CL_H",
+            strict=False,
+        )
+
+        suffix = {
+            "MT": "StaticMT",
+            "MTd": "StaticMTD",
+            "MD": "DynamicMT",
+            "MDd": "DynamicMT",
+        }.get(str(self.settings.compiler.runtime))
+
+        # GdiPlus requires C++, but ImageMagick has *.c files
+        project = (
+            "IM_MOD_emf_%s.vcxproj" % suffix
+            if self.options.shared
+            else "CORE_coders_%s.vcxproj" % suffix
+        )
+        tools.replace_in_file(
+            os.path.join("VisualMagick", "coders", project),
+            '<ClCompile Include="..\\..\\ImageMagick\\coders\\emf.c">',
+            '<ClCompile Include="..\\..\\ImageMagick\\coders\\emf.c">\n'
+            "<CompileAs>CompileAsCpp</CompileAs>",
+        )
+
+        for module in self._modules:
+            with tools.chdir(os.path.join("VisualMagick", module)):
+                msbuild = MSBuild(self)
+                msbuild.build(
+                    project_file="CORE_%s_%s.vcxproj" % (module, suffix),
+                    upgrade_project=False,
+                    platforms={"x86": "Win32", "x86_64": "x64"},
+                )
+
+        with tools.chdir(os.path.join("VisualMagick", "coders")):
+            pattern = (
+                "IM_MOD_*_%s.vcxproj" % suffix
+                if self.options.shared
+                else "CORE_coders_%s.vcxproj" % suffix
+            )
+            projects = glob.glob(pattern)
+            for project in projects:
+                msbuild = MSBuild(self)
+                msbuild.build(
+                    project_file=project,
+                    upgrade_project=False,
+                    platforms={"x86": "Win32", "x86_64": "x64"},
+                )
+
+    def _build_configure(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(
+            self, win_bash=tools.os_info.is_windows
+        )
+
+        # workaround for xorg/system adding systme includes https://github.com/conan-io/conan-center-index/issues/6880
```

Comment:
Let's hope this gets fixed.
```suggestion
        # FIXME: workaround for xorg/system adding systme includes https://github.com/conan-io/conan-center-index/issues/6880
```